src/containers/SellerProductsContainer.jsx

```diff
@@ -27,14 +35,64 @@ const SellerProductsContainer = () => {
     setItem(oneItem);
     return oneItem;
   };
+  const enableUpdate=()=>{
+    setUpdating(true)
+    setpoping(true);
+  }
+  const closingPopUp=()=>{
+    if(updating){
+      setUpdating(false)
+      setpoping(true)
+    }else{
+      setpoping(false)
+    }
+  }
+
+  const [updateProduct, {isSuccess, isError }] = useUpdateProductMutation();
+  const fileUploader = (file) => {
+    return new Promise((resolve, reject) => {
+      const fileReader = new FileReader();
+      fileReader.readAsDataURL(file);
+      fileReader.onload = () => {
+        resolve(fileReader.result);
+      };
+      fileReader.onerror = (err) => {
+        reject(err);
+      };
+    });
+  };
+  const updatingProduct=(async(data)=>{
+    const { image: uploadImages } = data;
+    
+    
+      const image = await Promise.all(
+        Array.from(uploadImages).map((file) => {
+          return fileUploader(file);
+        })
+      );
 
+    const modifiedData = {
+      ...data,
+      image
+    };
+    const {id} = items[0];
+    updateProduct({data: modifiedData, id})
+    if(!isError){
+      successNotification('Successfully updated product!')
+    }  
+    else{
+      ErrorNotification('Error in updating product!');
+    }
+    
```

Comment:
@frankflej, in a React component, it's generally recommended to separate concerns and keep functions focused on specific tasks. In this case, it's best to avoid placing code responsible for tracking success or errors within the update function.

Instead, consider the following approach:

1. Keep the update function responsible solely for updating the data.
2. Utilize React hooks to handle the tracking and evaluation of the update.

For example, you can use `useEffect` hook (for functional components) to monitor the success or errors of the data update. Within these methods, you can check the state or props to determine if the update was successful or if any errors occurred.

By separating these concerns, you'll have cleaner and more maintainable code that adheres to React's best practices.